### PR TITLE
Add k8s config REST endpoint and CLI command

### PR DESCRIPTION
**Pull request by bschimke95**
Provides an endpoint to retrieve an admin kubeconfig, e.g. for the charm. 
Note that this PR provides an initial implementation of the endpoint. As discussed, we might need argument overwrites (e.g. server address) in the future. For that, the `GetKubeConfigRequest` object can easily be extended later as needed.

The CLI is hidden and only a convenience for internal dev and test work (the same output can be obtained with `kubectl config view --raw`).

### CODE REVIEW

File: src/k8s/api/v1/cluster.go [hunk 1]
```diff
@@ -14,6 +14,14 @@ type InitClusterRequest struct{}
 // InitClusterResponse is the response for "POST 1.0/k8sd/cluster".
 type InitClusterResponse struct{}
 
+// GetKubeConfigRequest is used to ask for the admin kubeconfig
+type GetKubeConfigRequest struct{}
+
+// GetKubeConfigResponse is the response for "GET 1.0/k8sd/cluster/config".
+type GetKubeConfigResponse struct {
+	KubeConfig string `json:"kube_config"`
```
Comment (neoaggelos) [inline src/k8s/api/v1/cluster.go:22]: nit, `kubeconfig` is just fine

File: src/k8s/cmd/k8s/k8s_config.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+	"github.com/canonical/k8s/pkg/k8s/client"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	configCmd = &cobra.Command{
+		Use:    "config",
+		Short:  "Prints the admin config to connect to the cluster",
```
Comment (neoaggelos) [inline src/k8s/cmd/k8s/k8s_config.go:14]: nit, "Generate a kubeconfig that can be used to access the Kubernetes cluster"

File: src/k8s/pkg/k8sd/api/cluster_config.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/canonical/microcluster/state"
+)
+
+var k8sdClusterConfig = rest.Endpoint{
+	Path: "k8sd/config",
+	Get:  rest.EndpointAction{Handler: clusterConfigGet, AllowUntrusted: false},
+}
+
+func clusterConfigGet(s *state.State, r *http.Request) response.Response {
+	config, err := os.ReadFile("/etc/kubernetes/admin.conf")
```
Comment (neoaggelos) [inline src/k8s/pkg/k8sd/api/cluster_config.go:20]: should we instead render a new kubeconfig?

File: src/k8s/pkg/k8sd/api/cluster_config.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	"os"
+
+	apiv1 "github.com/canonical/k8s/api/v1"
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/state"
+)
+
+var k8sdClusterConfig = rest.Endpoint{
+	Path: "k8sd/config",
```
Comment (neoaggelos) [inline src/k8s/pkg/k8sd/api/cluster_config.go:15]: nit, `k8sd/kubeconfig`

File: src/k8s/pkg/k8sd/api/cluster_config.go (hunk 3 above)
Comment (neoaggelos) [reply src/k8s/pkg/k8sd/api/cluster_config.go:20]: not something for now, we could simply update the implementation later

**Review by neoaggelos — COMMENTED** [review]
Looks great, just minor nits really at the moment.

File: src/k8s/pkg/k8sd/api/cluster_config.go (hunk 3 above)
Comment (bschimke95) [reply src/k8s/pkg/k8sd/api/cluster_config.go:20]: Good point, I added a TODO

**Review by neoaggelos — APPROVED** [review]

**Pull request merged**